### ISSUE TITLE
SeriesGroupBy.apply raises AttributeError for '_contains_index_name'

### ISSUE THREAD

**Issue by bchu** [dask/dask#4590]
```
import dask.dataframe as dd
import pandas as pd
testdf = pd.DataFrame([[1, 2]], columns=['a', 'b'])
testdd = dd.from_pandas(testdf, npartitions=1).set_index('a')
testdd.b.groupby('a').apply(lambda d: d)
```

Results in the error:

```
... python3.6/site-packages/dask/dataframe/groupby.py in apply(self, func, *args, **kwargs)
   1112         df = self.obj
   1113         should_shuffle = not (df.known_divisions and
-> 1114                               df._contains_index_name(self.index))
   1115 
   1116         if should_shuffle:

AttributeError: 'Series' object has no attribute '_contains_index_name'
```

The error also occurs if I provide the meta object: `testdd.b.groupby('a').apply(lambda d: d, meta=('b', 'f8'))`.

**Comment by TomAugspurger** [comment]
https://github.com/dask/dask/blob/4bcd0fb6cc0add749d6f37b0bd86ef7dea1dab9d/dask/dataframe/groupby.py#L1079 seems to think that `self.obj` is always a dataframe, which may not always be correct.

If you use `testdd.groupby('a').b.apply(lambda d: d)`, then things work.

**Comment by jcrist** [comment]
Fixed by #4738.
